solr/core/src/java/org/apache/solr/core/SolrPaths.java

```diff
@@ -94,26 +87,6 @@ public static Path locateSolrHome() {
     return Paths.get(home).toAbsolutePath().normalize();
   }
 
-  public static void ensureUserFilesDataDir(Path solrHome) {
-    final Path userFilesPath = getUserFilesPath(solrHome);
-    final File userFilesDirectory = new File(userFilesPath.toString());
-    if (!userFilesDirectory.exists()) {
```

Comment:
The delta here with Paths API isn't line-for-line; it's much better.  `Files.createDirectories` works if it exists and simply throws if something goes wrong.